scipy/integrate/tests/test_cubature.py

```diff
@@ -872,6 +1260,45 @@ def test_genz_malik_1d_raises_error(self, xp):
             GenzMalikCubature(1, xp=xp)
 
 
+@array_api_compatible
+@skip_xp_backends(
+    "jax.numpy",
+    reasons=["transforms make use of indexing assignment"],
+)
+class TestTransformations:
+    @pytest.mark.parametrize(("a", "b", "points"), [
+        (
+            [0, 1, -math.inf],
+            [1, math.inf, math.inf],
+            [
+                [1, 1, 1],
+                [0.5, 10, 10],
+            ]
+        )
+    ])
+    def test_infinite_limits_maintains_points(self, a, b, points, xp):
+        """
+        Test that break points are correctly mapped under the _InfiniteLimitsTransform
+        transformation.
+        """
+
+        points = [xp.asarray(p, dtype=xp.float64) for p in points]
+
+        f_transformed = _InfiniteLimitsTransform(
+            # Bind `points` and `xp` argument in f
+            lambda x: f_with_problematic_points(x, points, xp),
+            xp.asarray(a, dtype=xp.float64),
+            xp.asarray(b, dtype=xp.float64),
+            xp=xp,
```

Comment:
@ollybritton here's the culprit for the array API failure. Either `xp` needs to come from `array_namespace` here, or `_InfiniteLimitsTransform` needs to apply `array_namespace` to the input `xp` during setup.

Reply:
Thanks!